Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/FlatPlatePaalmanPingsCorrection.py

```diff
@@ -504,83 +552,142 @@ def _flat_abs(self, angle):
+            acc += self._can_back_thickness * acc2 * np.exp(-ki_c1)
+            acc /= (self._can_front_thickness + self._can_back_thickness)
+        elif self._has_can_front_in:
+            acc = acc1
+        elif self._has_can_back_in:
+            acc = acc2
+
+        if self._has_sample_in:
+            if self._has_can_front_in and self._has_can_back_in:
+                acsc = self._can_front_thickness * acc1 * np.exp(-kf_s - kf_c2)
+                acsc += self._can_back_thickness * acc2 * np.exp(-ki_c1 - ki_s)
                 acsc /= (self._can_front_thickness + self._can_back_thickness)
+            elif self._has_can_front_in:
+                acsc = acc1 * np.exp(-kf_s)
+            elif self._has_can_back_in:
+                acsc = acc2 * np.exp(-ki_s)
+            else:
+                acsc = 0.5 * np.exp(-kf_s) + 0.5 * np.exp(-ki_s)
+        else:
+            acsc = acc
+
+        if self._has_can_front_in and self._has_can_back_in:
+            assc = ass * np.exp(-ki_c1 - kf_c2)
+        elif self._has_can_front_in:
+            assc = ass * np.exp(-ki_c1)
+        elif self._has_can_back_in:
+            assc = ass * np.exp(-kf_c2)
+        else:
+            assc = ass
 
-                assc = ass * np.exp(-ki_c1-kf_c2)
-            else:                                                          # reflection case
-                acc = self._can_front_thickness * acc1
-                acc += self._can_back_thickness * acc2 * np.exp(-ki_c1-kf_c1)
-                acc /= (self._can_front_thickness + self._can_back_thickness)
+        return assc, acsc, acc
 
+    # ------------------------------------------------------------------------------
+
+    def _container_reflection_calc(self, acc, acc1, acc2, ki_s, kf_s, ki_c1, kf_c1, ass):
+        if self._has_can_front_in and self._has_can_back_in:
+            acc = self._can_front_thickness * acc1
+            acc += self._can_back_thickness * acc2 * np.exp(-ki_c1 - kf_c1)
+            acc /= (self._can_front_thickness + self._can_back_thickness)
+        elif self._has_can_front_in:
+            acc = acc1
+        elif self._has_can_back_in:
+            acc = acc2
+
+        if self._has_sample_in:
+            if self._has_can_front_in and self._has_can_back_in:
                 acsc = self._can_front_thickness * acc1
-                acsc += self._can_back_thickness * acc2 * np.exp(-ki_c1-ki_s-kf_s-kf_c1)
+                acsc += self._can_back_thickness * acc2 * np.exp(-ki_c1 - ki_s - kf_s - kf_c1)
                 acsc /= (self._can_front_thickness + self._can_back_thickness)
+            elif self._has_can_front_in:
+                acsc = acc1
+            elif self._has_can_back_in:
+                acsc = acc2 * np.exp(-ki_s - kf_s)
+            else:
+                acsc = 0.5 + 0.5 * np.exp(-ki_s - kf_s)
+        else:
+            acsc = acc
+
+        if self._has_can_front_in and self._has_can_back_in:
+            assc = ass * np.exp(-ki_c1 - kf_c1)
```

Comment:
This is the second time np.exp(-ki_c1 - kf_c1) is calculated in this function ('_container_reflection_calc').
Is it unlikely that the factors ki_c1 and kf_c1 contain a large amount of data?

Reply:
As comment above.